pkg/webhooks/resource/utils.go

```diff
@@ -105,7 +105,11 @@ func skipBackgroundRequests(policy kyvernov1.PolicyInterface, logger logr.Logger
 	policyNew := policy.CreateDeepCopy()
 	policyNew.GetSpec().Rules = nil
 	for _, rule := range policy.GetSpec().Rules {
-		if rule.SkipBackgroundRequests && (bgsaDesired == bgsaActual) {
+		skipBackgroundRequests := true
```

Comment:
just try to track this [bug](https://github.com/kyverno/kyverno/issues/10421), why the `skipBackgroundRequests` is set to `true` by default here?

Reply:
https://github.com/kyverno/kyverno/blob/0bdbf7675e55210a8dd7690e9f80211e06ef41ab/api/kyverno/v1/rule_types.go#L106-L111

Reply:
Thanks @realshuting for the pointer.
I am trying to invest the issue we faced on the cluster policy cannot generate for existing resources in 1.12.5, I thought it is related to the `skipBackgroundRequests: true` by default (and it seems like [background scans](https://kyverno.io/docs/policy-reports/background/) is related to apply policy on existing resources). eventually, I found that it was related to this [issue](https://github.com/kyverno/kyverno/issues/10886).

However, I still want to know about the behaviour of `skipBackgroundRequests`, so far from my testing:
scenario: use the Kyverno cluster policy to generate resource 1 via rule 1 on an existing resource, and generate resource 2 based on resource 1 via rule 2.
1. resource 1 will be generated on existing resources regardless of the value of `skipBackgroundRequests`.
2. resource 2 will be generated only when `skipBackgroundRequests: false`.

I would like to know is 1 the expected behaviour? and is there any other scenario I need to set `skipBackgroundRequests: false`? (what in what scenario there are admission requests sent by the background controller), thanks.

If needed I can convert this to an issue.

Reply:
>I would like to know is 1 the expected behaviour?

Yes.

>what in what scenario there are admission requests sent by the background controller

For example, you have a generate policy creates a configmap and then a validate policy enforces configmaps settings, in this case, the validate policy is skipped by default (because of `skipBackgroundRequests: true`).

To illustrate the default setting of `skipBackgroundRequests`, you can image a generate policy is triggered by ConfigMap and also creates (clones) another Configmap, it will result in endless generation loop if not skipping those requests by default.

Reply:
thanks, it makes sense for your example.